packages/react-signature-frame/src/index.js

```diff
@@ -17,9 +17,26 @@ class AnvilSignatureFrame extends React.Component {
     window.removeEventListener('message', this.handleSignFinish)
   }
 
-  handleSignFinish = ({ origin, data: url }) => {
+  handleSignFinish = ({ origin, data }) => {
     if (this.props.anvilURL !== origin) return
-    this.props.onFinish(url)
+    if (typeof data === 'string') {
+      this.props.onFinish(data)
+
+      // parse query params into an object
+      const searchStr = data.split('?')[1]
+      const searchObj = JSON.parse('{"' + decodeURI(searchStr).replace(/"/g, '\\"').replace(/&/g, '","').replace(/=/g, '":"') + '"}')
```

Comment:
I think I'd do 2 things here that feel a little safer

1. Use [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) if it exists. You can do something like `if (typeof URLSearchParams !== 'undefined') {...}` then use it
2. Fallback to what you have. It also feels safer to decodeURI on each value component after you have the object, since often URLs will have their non-alpha characters encoded like `%20`

Reply:
FWIW, IE doesnt support URLSearchParams, so IE users would fallback to the json parse

Reply:
Implemented